### ISSUE TITLE
replacement for query.from_statement for 2.0

### ISSUE THREAD

**Issue by zzzeek** [sqlalchemy/sqlalchemy#5293]
copying some notes to see how from statement will work without using session.query()

```
    # 3. thoughts on from statement / text.... the ability to "give me ORM
    # entities from prefab statement" use case does not fit well into any of
    # this so we might need a new construct to make this clear / explicit:

    a1 = aliased(A)
    a2 = aliased(A)

    stmt = text(
        "select a1.id, a1.data, a2.id, a2.data FROM a AS a1, a AS a2"
    ).columns(a1.id, a1.data, a2.id, a2.data).\
        select_entities(a1, a2).options(contains_eager(a1.bs))


    stmt = select(a1.id, a1.data, a2.id, a2.data).select_entities(a1, a2)

    # can we do this?  does this modify the query?  or have we given it
    # specific columns to select, therefore we are in "no modify" mode?

    stmt = select(a1.id, a1.data, a2.id, a2.data).select_entities(a1, a2).\
        options(joinedload(a1.bs))

    # but then what if we did this?  what does that mean ?
    stmt = select(a1, a2.id, a2.data).select_entities(a1).\
        options(joinedload(a1.bs))

    # it would mean....if it's a "select()" we would make all the ORM entities
    # like we do, apply them to build the SELECT string, then...do a
    # from_statement on it.

    # I think select_entities() is a new construct.   Sorry, it just is.

    stmt = select(a1, a2.id, a2.data)

    stmt = select_entities(a1).from_statement(stmt).options(joinedload(a1.bs))

    # can we make this thing work without QueryCompileState?  can we do
    # row_processor() etc without the entity objects being part of a compile
    # state etc.?

    # applies to core...it should?  somehow?
    # or just have it as sqlalchemy.future.orm.select_entities for now?
    # what happens when you send select_entities() to the compiler?
    # if no statement, raise CompileError
    # if statement, compile the statement, do everything as though that
    # statement were it.   then apply .columns() to the result?  this means
    # new thingy on SQLCompiler to indicate this?    how does the ORM make use
    # of this?   is this really useful?

    stmt = select(table.c.a, table.c.b, table.c.c).select_entities(table.c.b)


    # is equivalent to ?  can we apply row processors
    # after execute?

    stmt = select(table.c.a, table.c.b, table.c.c)
    result = thing.execute(stmt).columns(table.c.b)

```

**Comment by zzzeek** [comment]
note that this is currently available with the ``FromStatement`` construct that can be seen at https://github.com/sqlalchemy/sqlalchemy/blob/master/lib/sqlalchemy/orm/query.py#L3406 .

My current plan for this is to just add a ``from_statement()`` "extension" method right on to the base ``select()``.   This method will make use of the plugin approach we are using for "column descriptions" and a few other methods right now.   the method is only meaningful for a statement that has acquired the "orm" plugin from its arguments.

   strsql = text("select * from user").columns(User.id, User.name)
   stmt = select(User).from_statement(strsql)
   users = session.execute(stmt).all()

**Comment by zzzeek** [comment]
this is merged in 6ee643d723e8d65fb4bd3c8848b70693966ff3e5 https://github.com/sqlalchemy/sqlalchemy/blob/6ee643d723e8d65fb4bd3c8848b70693966ff3e5/lib/sqlalchemy/sql/selectable.py#L4323
